### DISCUSSION TITLE
TX23 Anemometer

### DISCUSSION THREAD

**Discussion by Shumde** [arendst/Tasmota#11155] [Q&A] [answered]
Hi, I installed Tasmota with TX23 - works fine. Now I want to make a Rule.
But I don´t know which name for the trigger I have to use.

What I want is: Sending per WebSend WindSpeed Act and Winddirection to my Homematic-System.
Http-Call is working - just the right trigger is missing.

Thank You 
Shumde

![TX23](https://user-images.githubusercontent.com/12113387/109418282-e907fe00-79c7-11eb-8b0b-0e62bbbb16b6.png)

**Comment by javier6d2** [comment, marked as answer]
I use a TX20 and publish data to Domoticz with MQTT as follows, may be you can use similar code to HA:

Rule1
on tele-TX20#Speed#Avg do var1= 5.14444444*%value% endon
on tele-TX20#Speed#Max do var2= 5.14444444*%value% endon
on tele-TX20#Dir#Avg do var3 %value% endon
on tele-TX20#Dir#AvgCard do publish domoticz/in {"idx":295,"svalue":"%var3%;%value%;%var1%;%var2%;0;0"} endon

**Important:**

1) Domoticz only ACCEPTS wind data input units: 10*m/s, so I have converted knots (my wind sensor unit) to 10*m/s with 5.14444 factor.
2) To be able convert data to 10*m/s you need compile your own tasmota firmare, check: https://tasmota.github.io/docs/Rules/
Add the following to user_config_override.h:
`#define USE_EXPRESSION // Add support for expression evaluation in rules (+3k2 code, +64 bytes mem)`

It is easy to publish from script also, see scripting doc. Driver is not compatible with rules.

**Reply by barbudor** [reply]
@javier6d2 There is something wrong with this syntax 
```
on tele-TX20#Speed#Avg do var1= 5.14444444*%value% %value% endon
````
`%value%` should probably be there only once

**Reply by Shumde** [reply]
Thank you!
In the meantime i studied the rules und tried something - and it works!
I think your code will help too.

**Reply by javier6d2** [reply]
> `%value%` should probably be there only once

@barbudor thank you, you are right
